tests/test_osm_client.py

```
@@ -0,0 +1,135 @@
+from custom_components.places.osm_client import OSMClient
+
+
+class AioClientMock(Protocol):
+    """Minimal aiohttp mock interface used by these tests."""
+
+    def get(self, url: str, **kwargs: object) -> object:
+        """Register mocked responses for URL requests."""
+
+
+@pytest.mark.asyncio
+async def test_reverse_url_matches_nominatim_query_contract(mock_hass: HomeAssistant) -> None:
```

Ruff (0.15.12)

[warning] Unused function argument: `mock_hass`

(ARG001)